src/test/unit/auth.spec.ts

```diff
@@ -0,0 +1,63 @@
+import * as jestMock from "jest-mock";
+import expect from "expect";
+// vscode mock must be imported before the tested module!!!
+import {
+  contextMock,
+  executeRegisteredCommand,
+  mockLogin,
+  resetVscodeMocks,
+} from "./mocks/vscode";
+import { AuthService } from "../../auth";
+import { mockValidateCredentialsRequest } from "./mocks/http";
+
+describe("authentication", () => {
+  afterEach(() => {
+    resetVscodeMocks();
+  });
+
+  it("should save login credentials in secrets store", async () => {
+    const keyId = "keyId";
+    const secret = "secret";
+    const onLogin = jestMock.fn();
+    const auth = new AuthService(contextMock);
+    auth.registerLoginCommand(onLogin);
+
+    mockLogin(keyId, secret);
+    mockValidateCredentialsRequest({ keyId, secret });
+    await executeRegisteredCommand("env0.login");
+
+    expect(onLogin).toHaveBeenCalled();
+    expect(contextMock.secrets.store).toHaveBeenCalledWith("env0.keyId", keyId);
+    expect(contextMock.secrets.store).toHaveBeenCalledWith(
+      "env0.secret",
+      secret
+    );
+  });
+
+  it("should not save login credentials in secrets store when credentials invalid", async () => {
```

Comment:
You can DRY these log in tests I see a lot of repeated code here. Maybe add a `describe` section for login and run the common setup in `beforeEach` or something like that

Reply:
c6e97ff646c56acdcd8bf1913c33df8afab3b002